datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -290,6 +290,19 @@ def service_metadata(self, meta_name, value):
     def check(self, instance):
         raise NotImplementedError
 
+    def set_external_tags(self, external_tags):
+        from datadog_agent import set_external_tags as _set_external_tags
+        # Example of external_tags format
+        # [
+        #     ('hostname', {'src_name': ['test:t1']}),
+        #     ('hostname2', {'src2_name': ['test2:t3']})
+        # ]
+        for x in external_tags:
+            for src_name, tags in iteritems(x[1]):
```

Comment:
Should we validate the structure of `x` just in case it is not set properly?

Reply:
I'll wrap in a try/catch around an indexError, what do you think?

Reply:
Actually, discussed this with @zippolyte The Agent currently  emits an error if the format of these tags isn't correct. In that spirit, I've set this to catch an IndexError and log an exception (and then raise that exception) This keeps the same functionality that the Agent was currently performing.